int32 lowX
int32 lowY
int32 highX
int32 highY
int32 r
int32 g
int32 b
int32 thickness